src/TLuaInterpreter.cpp

```diff
@@ -5479,223 +5477,238 @@ int TLuaInterpreter::tempTimer(lua_State* L)
         lua_pushstring(L, "tempTimer: wrong argument type");
         lua_error(L);
         return 1;
-    } else {
-        luaFunction = lua_tostring(L, 2);
     }
 
+    QString luaCodeAsString = QString::fromUtf8(lua_tostring(L, 2));
+
     Host& host = getHostFromLua(L);
     TLuaInterpreter* pLuaInterpreter = host.getLuaInterpreter();
-    QString _fun = luaFunction.c_str();
-    int timerID = pLuaInterpreter->startTempTimer(luaTimeout, _fun);
+    int timerID = pLuaInterpreter->startTempTimer(luaTimeout, luaCodeAsString);
     lua_pushnumber(L, timerID);
     return 1;
 }
 
 int TLuaInterpreter::tempExactMatchTrigger(lua_State* L)
 {
-    string luaRegex;
+    Host& host = getHostFromLua(L);
+    TLuaInterpreter* pLuaInterpreter = host.getLuaInterpreter();
+    QString exactMatchPattern;
+    int triggerID;
+
     if (!lua_isstring(L, 1)) {
-        lua_pushstring(L, "tempExactMatchTrigger: wrong argument type");
-        lua_error(L);
-        return 1;
-    } else {
-        luaRegex = lua_tostring(L, 1);
+        lua_pushfstring(L, "tempExactMatchTrigger: bad argument #1 type (exact match pattern as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
     }
 
-    string luaFunction;
-    if (!lua_isstring(L, 2)) {
-        lua_pushstring(L, "tempExactMatchTrigger: wrong argument type");
-        lua_error(L);
-        return 1;
+    exactMatchPattern = QString::fromUtf8(lua_tostring(L, 1));
+
+    if (lua_isstring(L, 2)) {
+        QString luaFunction = QString::fromUtf8(lua_tostring(L, 2));
+        triggerID = pLuaInterpreter->startTempExactMatchTrigger(exactMatchPattern, luaFunction);
+    } else if (lua_isfunction(L, 2)) {
+        triggerID = pLuaInterpreter->startTempExactMatchTrigger(exactMatchPattern, QString());
+
+        auto trigger = host.getTriggerUnit()->getTrigger(triggerID);
+        trigger->mRegisteredAnonymousLuaFunction = true;
+        lua_pushlightuserdata(L, trigger);
+        lua_pushvalue(L, 2);
+        lua_settable(L, LUA_REGISTRYINDEX);
     } else {
-        luaFunction = lua_tostring(L, 2);
+        lua_pushfstring(L, "tempExactMatchTrigger: bad argument #2 type (code to run as a string or a function expected, got %s!)", luaL_typename(L, 1));
```

Comment:
`luaL_typename(L, 1)` should be `luaL_typename(L, 2)`. This is wrong in a lot of functions.